### ISSUE TITLE
Using .select([]) to filter out rows is broken and removes the scope of user in this example

### ISSUE THREAD

**Issue by steipete** [groue/GRDB.swift#965]
Two variants of the query. In the second one using the query language, `.select([])` breaks the query as the scope is removed.

```swift
#if false
let sql: SQLRequest<TweetInfo> = """
SELECT
    \(columnsOf: Tweet.self), -- instead of tweet.*
    \(columnsOf: User.self) -- instead of user.*
FROM
    tweetUserFavorite
    JOIN tweet ON tweet.id = tweetUserFavorite.tweetId
    JOIN user ON user.id = tweet.userId
WHERE
    tweetUserFavorite.userId == \(userId)
ORDER BY
    tweet.createdAt DESC
LIMIT \(limit)
"""
let request = sql.adapted { db in
    let adapters = try splittingRowAdapters(columnCounts: [Tweet.numberOfSelectedColumns(db)])
    return ScopeAdapter(["user": adapters[1]])
}
return try request.fetchAll(db)
#endif

return try TweetUserFavorite
    // GRDB bug: Select nothing doesn't work, removes scope?
    //.select([])
    .filter(TweetUserFavorite.Columns.userId == userId)
    .including(required: TweetUserFavorite.tweet
                .if(tweetUserId != nil) { request in
                    request
                        .filter(Tweet.Columns.userId == tweetUserId!)
                }
                .order(Tweet.Columns.createdAt).reversed())
    .including(required: TweetUserFavorite.user)
    .asRequest(of: TweetInfo.self)
    .limit(limit)
    .fetchAll(db)
```

Code 12.5GM, GRDB latest as of April 22, 2021.

Example project shared privately with @groue.

**Comment by groue** [comment]
Thanks for the report @steipete!

Yes indeed `MyRecord.select([]).including(...)` looks broken. I'll look into it.

**Comment by groue** [comment]
The fix is ready in #974. It will ship in the next version!

**Comment by groue** [comment]
Fixed in v5.8.0
